src/resolution/instantiation.ts

```diff
@@ -29,45 +25,121 @@ function _injectProperties(
 
 }
 
-function _createInstance(Func: interfaces.Newable<any>, injections: Object[]): any {
-    return new Func(...injections);
+function _createInstance(
+    constr: interfaces.Newable<any>,
+    childRequests: interfaces.Request[],
+    resolveRequest: interfaces.ResolveRequestHandler,
+): any {
```

Comment:
if we manage to type this, all the work forward to replace any's will be so easy 💯